### ISSUE TITLE
nightly red: jax 0.11.2 makes test_bug_1661's orphan-atime reproduction vacuous (DID NOT RAISE)

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#2416]
Main's nightlies of 09-18 and 09-19 (runs 35326825041, 35432700576, both at `0eab59b60`) fail `test (regression-b1, 1, 1, 50)` on

```
tests/regression/bug/test_bug_1661_orphan_atime_kills_cache_writes.py::test_orphan_breaks_a_real_jax_put_and_the_repair_fixes_it
Failed: DID NOT RAISE FileNotFoundError
```

and every PR whose `pull_request` run built after ~09-18 08:00 UTC inherits the identical red on its merge commit.

## Root cause: jax 0.11.2, not the repo

The verdict tracks the installed jax version, not the code:

| run | date (UTC) | repo SHA | jax installed | regression-b1 |
|---|---|---|---|---|
| 35204533615 (nightly) | 09-17 09:20 | bbdd72cb6 | **0.11.1** | success |
| PR #2414 merge-commit run | 09-18 ~01:30 | 0eab59b60 + trivial diff | (cache-restored pre-release env) | success |
| 35326825041 (nightly) | 09-18 08:55 | 0eab59b60 | **0.11.2** | failure |
| 35432700576 (nightly) | 09-19 08:42 | 0eab59b60 | **0.11.2** | failure |

(Versions read from the jobs' pip logs.) jax 0.11.2 was released 2026-09-17T23:41 UTC — squarely between the last green and first red nightly. Locally, the full test file passes 17/17 at origin/main (`52c356a31`) on jax 0.11.1. The five commits in the `bbdd72cb6..0eab59b60` window touch the data locator, guards, and parameters — nothing on the JAX-cache path.

## What the failure means

The test (from #1662, the #1661 repair) reproduces the disease first: an orphaned `-atime` marker must make a **real** `jax` compilation-cache `put` raise `FileNotFoundError`, and the repair must then fix it. On jax 0.11.2 the put **no longer fails** under the orphan condition — upstream changed cache-write behavior (more tolerant write path), so the reproduction arm is vacuous: DID NOT RAISE.

Note the repair itself is not shown broken — what broke is the ability to reproduce the original disease on the new jax.

## Fix options (decision needed, in preference order)

1. **Verify-then-gate**: confirm on jax 0.11.2 that the orphan condition is truly harmless upstream (put succeeds AND the cache entry is actually written and readable — not silently skipped, which would be the same disease with a quieter symptom). If harmless, gate the reproduction arm on `jax.__version__ < "0.11.2"` (skip with a reason naming this issue), keeping the repair + its unit-level tests active — the repair still matters for users on older jax.
2. If the 0.11.2 put *silently drops* the write instead of raising, the disease is alive with a worse symptom: the test needs to assert the outcome (entry present) rather than the exception, and the repair stays load-bearing.
3. Pinning jax in CI is a policy call above this issue — it would also have masked the finding.

Option 1's verification step decides between 1 and 2 — do not gate on version without it (a probe that can't see silent drops proves nothing).

Unclaimed — free for any session; claim by comment before starting.

**Comment by suchethac** [comment]
Claiming from the "review tengri issues" fix wave: a measure-first lane reproduces the 0.11.2 behaviour in an isolated venv, establishes whether the orphan-atime put is harmless (entry actually written) or silently dropped, and fixes the test's reproduction arm accordingly — asserting the outcome in both jax versions, never skipping. PR to follow.
